test/Benchmarks/src/Vector/Vector.enso

```diff
@@ -24,8 +24,11 @@ make_random_vec n =
 main =
     random_vec = make_random_vec vector_size
     random_vec_2 = make_random_vec 100000
+    random_gen = Random.new 123456
 
     Bench.measure (Base.Vector.new vector_size i->i) "New Vector" iter_size num_iterations
+    Bench.measure (Base.Vector.fill vector_size 42) "Fill Constant" iter_size num_iterations
+    Bench.measure (Base.Vector.fill vector_size random_gen.nextLong) "Fill Random" iter_size num_iterations
```

Comment:
I'm wondering if this kind of usecase should be supported by `fill`. Why not use `Vector.new vector_size _->random_gen.nextLong` then? (I know with the new version this is exactly what `fill` will call).

My point is - I'm wondering why `fill` has `~item` suspended so it's recomputed on each invocation.

IMO what `fill` suggests is that it should exactly fill an array with a constant, not recompute on each pass. What do you think? Either way I think we have to update the doc comment as it's completely out of date - it even has wrong argument names there (`constructor` vs `item`).

The doc says about filling a vector with a **constant**. I prefer that and I'd suggest to remove the `~` from `item` to indeed compute the constant only once. Then we should probably add a test that checks that the `item` is _not_ recomputed many times.

But if you think we should keep the `~`, I'm also fine with that - but then we really _need_ to remove the word 'constant' from `fill`'s doc, because it's only confusing.

Reply:
Removed the`~` and added new tests too (one constant, one now randoms)

Reply:
Perfect, thanks!

Can we add a test to `Vector_Spec.enso`?:

```diff
     Test.specify "should allow vector creation with a constant constructor" <|
         Vector.fill 100 1 . fold (0) (+) . should_equal 100
+        r = Ref.new 0
+        next =
+            r.put r.get+1
+        mut = Vector.new 4 _->next
+        mut.should_equal [0, 1, 2, 3]
+
+        const = Vector.fill 4 next
+        const.should_equal [4, 4, 4, 4]
```